### ISSUE TITLE
The redirect API descriptions are incorrect

### ISSUE THREAD

**Issue by billti** [AzureAD/microsoft-authentication-library-for-js#4377]
### Core Library

MSAL.js v2 (@azure/msal-browser)

### Wrapper Library

Not Applicable

### Documentation Location

lib\msal-browser\src\app\ClientApplication.ts

### Description

The comments/docs for the APIs such as loginRedirect and acquireTokenRedirect state

> This function redirects the page, so any code that follows this function will not execute.

However this is not true. I was getting exceptions on the following lines expecting them to not execute per the docs, but they were running after the API call.

It looks like these functions actually return a Promise, so if you await the promise then that will never return, but code you have on following lines that doesn't await will execute.

e.g. Code snippet from my sample app

```js
                msalInstance.loginRedirect();
                console.error("On the line after loginRedirect!!!");
```

Output in the console with verbose logging on:

```txt
MSAL: Level: 3: [Sat, 08 Jan 2022 06:19:52 GMT] : @azure/msal-browser@2.21.0 : Verbose - loginRedirect called
VM10:147 MSAL: Level: 3: [Sat, 08 Jan 2022 06:19:52 GMT] : @azure/msal-browser@2.21.0 : Verbose - preflightBrowserEnvironmentCheck started
VM10:147 MSAL: Level: 3: [Sat, 08 Jan 2022 06:19:52 GMT] : @azure/msal-browser@2.21.0 : Verbose - acquireTokenRedirect called
VM10:147 MSAL: Level: 3: [Sat, 08 Jan 2022 06:19:52 GMT] : @azure/msal-browser@2.21.0 : Verbose - getAllAccounts called
VM10:147 MSAL: Level: 2: [Sat, 08 Jan 2022 06:19:52 GMT] : @azure/msal-browser@2.21.0 : Info - Emitting event: msal:loginStart
VM10:147 MSAL: Level: 3: [Sat, 08 Jan 2022 06:19:52 GMT] : [45a099c2-d35e-48a7-ad84-d7cd7dd400a0] : msal.js.browser@2.21.0 : Verbose - preflightInteractiveRequest called, validating app environment
VM10:147 MSAL: Level: 3: [Sat, 08 Jan 2022 06:19:52 GMT] : [45a099c2-d35e-48a7-ad84-d7cd7dd400a0] : msal.js.browser@2.21.0 : Verbose - initializeAuthorizationRequest called
VM10:147 MSAL: Level: 3: [Sat, 08 Jan 2022 06:19:52 GMT] : [45a099c2-d35e-48a7-ad84-d7cd7dd400a0] : msal.js.browser@2.21.0 : Verbose - getRedirectUri called
VM10:147 MSAL: Level: 3: [Sat, 08 Jan 2022 06:19:52 GMT] : [45a099c2-d35e-48a7-ad84-d7cd7dd400a0] : msal.js.browser@2.21.0 : Verbose - Initializing BaseAuthRequest
VM10:147 MSAL: Level: 3: [Sat, 08 Jan 2022 06:19:52 GMT] : [45a099c2-d35e-48a7-ad84-d7cd7dd400a0] : msal.js.browser@2.21.0 : Verbose - Authentication Scheme wasn't explicitly set in request, defaulting to "Bearer" request
VM10:170 
        
       On the line after loginRedirect!!!
(anonymous) @ VM10:170
Promise.then (async)
requireAccount @ VM10:160
(anonymous) @ VM10:1741
(anonymous) @ VM10:1756
(anonymous) @ VM10:1758
VM10:147 MSAL: Level: 3: [Sat, 08 Jan 2022 06:19:52 GMT] : @azure/msal-browser@2.21.0 : Verbose - loginRedirect called
VM10:147 MSAL: Level: 3: [Sat, 08 Jan 2022 06:19:52 GMT] : @azure/msal-browser@2.21.0 : Verbose - preflightBrowserEnvironmentCheck started
VM10:147 MSAL: Level: 3: [Sat, 08 Jan 2022 06:19:52 GMT] : @azure/msal-browser@2.21.0 : Verbose - acquireTokenRedirect called
VM10:147 MSAL: Level: 3: [Sat, 08 Jan 2022 06:19:52 GMT] : @azure/msal-browser@2.21.0 : Verbose - getAllAccounts called
VM10:147 MSAL: Level: 2: [Sat, 08 Jan 2022 06:19:52 GMT] : @azure/msal-browser@2.21.0 : Info - Emitting event: msal:loginStart
VM10:147 MSAL: Level: 3: [Sat, 08 Jan 2022 06:19:52 GMT] : [5f3e3a2e-6584-4236-a2f6-58b22ec05d1b] : msal.js.browser@2.21.0 : Verbose - preflightInteractiveRequest called, validating app environment
VM10:147 MSAL: Level: 3: [Sat, 08 Jan 2022 06:19:52 GMT] : [5f3e3a2e-6584-4236-a2f6-58b22ec05d1b] : msal.js.browser@2.21.0 : Verbose - initializeAuthorizationRequest called
VM10:147 MSAL: Level: 3: [Sat, 08 Jan 2022 06:19:52 GMT] : [5f3e3a2e-6584-4236-a2f6-58b22ec05d1b] : msal.js.browser@2.21.0 : Verbose - getRedirectUri called
// etc,,
```

**Comment by tnorling** [comment]
@billti You're right, because these APIs are asynchronous that comment should probably read: `This function redirects the page, so any code that follows this function should not be expected to execute.`

**Comment by billti** [comment]
Thanks @tnorling. To me that still implies that nothing will execute after this call, which isn't the case. I think a more accurate statement would be `This function initiates a page redirect. The Promise returned by this call will never complete, however any subsequent code that doesn't "await" that promise may continue to execute until the redirect occurs`

Or is that a little too wordy and technical?

**Comment by tnorling** [comment]
@billti Feel free to open a pull request with your proposed changes
